utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -673,6 +676,57 @@ where
     }
 }
 
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+pub struct CodePointMapRange {
```

Comment:
Suggestion: Let's brainstorm better names for this type.  It is really a quite general type; it has nothing itself to do with the fact that it comes from a CodePointMap.  I see it as a code point range with a value.  Maybe CodePointRangeWithValue?  That's a mouthful.  Something better?

Reply:
I would keep the more specific name, despite the fairly generic set of fields. I think it would be confusing if unrelated crates used the same type (and had a dependency to it) just because it happens to have the same set of fields. Conversely, keeping this name makes it more obvious that it's the type for the functions here.

Reply:
An argument for making a more generic type is so that we could have a function to convert from `impl Iterator<Item = CodePointMapRange>` to `UnicodeSet`.  In order to have such a function, the type should be defined in the `UnicodeSet` crate.

Why not just make it a bare tuple?  Just `(Range<u32>, T)`

Reply:
Actually, I think:

1. `CodePointTrie::iter_ranges()` should return `impl Iterator<Item = (Range<u32>, T)>`
2. `ScriptExtensions::iter_for_script()` should return `impl Iterator<Item = Range<u32>>`
3. There should be a function `UnicodeSet::from_range_iterator()` that takes the above iterator type
4. We could still have `ScriptExtensions::create_set_for_script()` that calls the two above functions for convenience

Reply:
What should the `UnicodeSet::from_range_iterator()` do with the value? UnicodeSet should only see "true" ranges. Some other piece of code should filter and transform CodePointTrie ranges into UnicodeSet ranges.

Reply:
`UnicodeSet::from_range_iterator()` should take a value-less range.

Note that my suggested `ScriptExtensions::iter_for_script()` returns a value-less iterator, since it's implied that all ranges have the same value (the script that was passed as an argument to that function).  The implementation of that function should essentially be `self.cpt.iter_ranges().filter(...).map(...)` to get the iterator to the correct form.

Reply:
Ok, I missed the Iterator signature change. Fine by me.

Reply:
That sounds good to me.

When talking with Shane about the return value for the API that returns the ranges where Script_Extensions matches for a given Script code (`get_script_extension_ranges`), that is one subset of the general case in which the return value range can be a Rust-y Range -- just a start point and end point -- because the CodePointMap value is implied/already correlated to the API+input values, so it is no longer needed.

So more concretely, currently we have `pub fn get_script_extensions_ranges(&self, script: Script) -> impl Iterator<Item = RangeInclusive<u32>> ...`

But for the general case, `CodePointMapRangeIterator` still provides an iterator of `CodePointMapRange`s that still represent a value + range.